withReduxSaga/index.js

```diff
@@ -0,0 +1,45 @@
+import createSagaMiddleware, { END } from 'redux-saga';
+import withSaga from './withReduxSaga';
+
+export default (config, InnerComponent, actions) => {
+  const middlwareOptions = config.sagaMonitor ? { sagaMonitor: config.sagaMonitor } : {};
+  const sagaMiddleware = createSagaMiddleware(middlwareOptions);
+
+  const configureStore = config.configureStore(sagaMiddleware);
+
+  const STORE_KEY = '_NEXT_STORE_KEY_';
+  const existingStore = (global.window && global.window[STORE_KEY]);
+  const params = {
+    store: existingStore || {},
+  };
+  params.makeStore = {
+    createStore: (initialState, options) => {
+      params.store = configureStore(initialState, options);
```

Comment:
`params.makeStore.createStore` is called by `withRedux` with the `initialState` it has built up. I intercept that store building so I can get the reference to the store. This does give us a side effect of having an opportunity to add additional state to the `configureStore` function. I'm just not sure what we would add.

I guess I could add an optional parameter that identifies some initial state. My gut feeling however is to let the saga handle setting default state.